pkg/webhooks/server.go

```diff
@@ -459,12 +460,20 @@ func (ws *WebhookServer) resourceValidation(request *v1beta1.AdmissionRequest) *
 	if err != nil {
 		logger.Error(err, "failed to load incoming request in context")
 	}
+	resource, err := utils.ConvertResource(request.Object.Raw, request.Kind.Group, request.Kind.Version, request.Kind.Kind, request.Namespace)
+
+	obj := resource.UnstructuredContent()
 
 	err = ctx.AddUserInfo(userRequestInfo)
 	if err != nil {
 		logger.Error(err, "failed to load userInfo in context")
 	}
 
+	err = ctx.AddImageDetails(obj["kind"], obj["spec"])
+	if err != nil {
+		logger.Error(err, "failed to load image details in context")
+	}
```

Comment:
In order to avoid serializing / deserializing, you can add image details in [HandleValidation](https://github.com/kyverno/kyverno/blob/274287617a752c4822a13acaee96339da3eb9a75/pkg/webhooks/validation.go#L30), before we build policy context: 
https://github.com/kyverno/kyverno/blob/274287617a752c4822a13acaee96339da3eb9a75/pkg/webhooks/validation.go#L74-L75